sdk/src/beta9/abstractions/taskqueue.py

```diff
@@ -230,26 +231,35 @@ def _handle_serve_interrupt(self) -> None:
         os._exit(0)  # kills all threads immediately
 
     def _serve(self, *, dir: str, object_id: str):
-        with ThreadPoolExecutor() as pool:
```

Comment:
I replaced the pool executor because the thread for `self.parent.sync_dir_to_workspace` never shut down and blocked termination of the application.